### DISCUSSION TITLE
Set hostname of box from fritz.box to 3cx.eu to achieve split DNS

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#893] [Allgemein]
3CX has deprecated the local login with an IP.
They want the lan phone devices to login to the 3cx server with the same FQDN that's used for logging in from the Internet.

For this you need split-DNS or a host override
Alas the FB does not support this.

I almost thought I could abuse CONFIG_HOSTNAME
```
CONFIG_HOSTNAME=3cx.eu

ctlmgr -s
multid -s
ctlmgr
multid
```
Well, actually I can...

if **ctlmgr** and **multid** run with the variable CONFIG_HOSTNAME set, it will assign all hosts a FQDN within that domain.
I still need to find out the most elegant way to set that variable.

```
root@fritz:# nslookup frater.3cx.eu
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost

Name:      frater.3cx.eu
Address 1: 192.168.188.90 frater.3cx.eu
root@fritz:/var/flash# nslookup 192.168.188.90
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost

Name:      192.168.188.90
Address 1: 192.168.188.90 frater.3cx.eu

```

I thought I had my proof of concept, but needed to do another test...  a test that failed.
It can't resolve any "real" 3cx.eu addresses anymore as they are all searched locally.

Let's pretend that acme.3cx.eu is a valid foreign 3CX-system.
```
root@fritz:# nslookup acme.3cx.eu
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost

nslookup: can't resolve 'acme.3cx.eu': Name or service not known

```
I could still use it, but it's extremely messy if people come in with their smartphone using their own 3cx.eu FQDN.
That's not how a host-override would work. It would still resolve the other FQDN's of 3cx.eu
On the other hand those people shouldn't be on that LAN anyway.

Any other thoughts?
I could of course switch to **dnsmasq** which I never did on a Fritz!box.
Does it work good?

I have some time before I am forced to use split-DNS for 3CX.
It's a shame it isn't supported natively.

Someone tried this before and already knows an elegant way to set the hostname?

I've also read this https://www.ip-phone-forum.de/threads/kann-dns-domain-nicht-%C3%A4ndern.285020/ 
but couldn't get it to work.
It probably doesn't work anymore with newer versions and it now only supports toplevels (`without embedded dots`) which makes it completely useless.

**Comment by fraternl** [comment]
I did a `find -type f -exec grep -H CONFIG_HOSTNAME {} \;` and now discovered it is already in FREETZ

![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/0297a563-b1f1-4acd-a897-1434ea35fbef)

```
nslookup frater.3cx.eu
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost

Name:      frater.3cx.eu
Address 1: 192.168.188.90 frater.3cx.eu

```
It's however all locally on the box itself ;-(

**Comment by fda77** [comment]
Simply change the hostname is possible by the AVM-webif. But it changes also many other things like AP-name, Mediaserver etc and Freetz does it not. I'm using since a long time  a hostname with dots with is only possible by ar7 editiong as ctlmgr_ctl deny updates with a dot..

**Comment by fraternl** [comment]
I don't mind about the mediaserver, nas and myfritz.  I don't use them.

The caveat of not being able to resolve any other 3cx.eu domain is the reason I'm dismissing it.
It would have worked otherwise.

BTW box is a valid TLD and the name fritz.box is not registered there.
I wonder what kind of can of worms it opens up if such a domain is registered.....

```
whois box
% IANA WHOIS server
% for more information on IANA, visit http://www.iana.org
% This query returned 1 object

domain:       BOX

organisation: Intercap Registry Inc.
address:      103 South Church Street, 4th Floor, Harbour Place, PO Box 10240
address:      George Town Grand Cayman KY1-1002
address:      Cayman Islands (the)

contact:      administrative
name:         General Manager
organisation: Intercap Registry Inc.
address:      4th Floor, Harbour Place, 103 South Church Street
address:      PO Box 10240
address:      Grand Cayman KY1-1002
address:      Cayman Islands (the)
phone:        +1 833 436 8462
e-mail:       admin@intercap.inc

contact:      technical
name:         CTO
organisation: CentralNic
address:      Saddlers House, 4th Floor
address:      44 Gutter Lane
address:      London EC2V 6BR
address:      United Kingdom of Great Britain and Northern Ireland (the)
phone:        +44 20 33 88 0600
fax-no:       +44 20 33 88 0601
e-mail:       tld.ops@centralnic.com

nserver:      A.NIC.BOX 194.169.218.139 2001:67c:13cc:0:0:0:1:139
nserver:      B.NIC.BOX 185.24.64.139 2a04:2b00:13cc:0:0:0:1:139
nserver:      C.NIC.BOX 212.18.248.139 2a04:2b00:13ee:0:0:0:0:139
nserver:      D.NIC.BOX 212.18.249.139 2a04:2b00:13ff:0:0:0:0:139
ds-rdata:     63242 8 2 f63250257503cce60195e71c2e346a2d8bde06dcbd666b9ee4f860d75988702d
ds-rdata:     62294 8 2 29a911bdbf1fc105d593ea39a619bf1e89cb1bfe73ff70fb7e80b75ab9a7a8b8

whois:        whois.nic.box

status:       ACTIVE
remarks:      Registration information: https://intercap.inc

created:      2016-10-27
changed:      2023-11-17
source:       IANA

```

**Comment by fda77** [comment]
my hostname is "device.xxx" and when i ping "other" it resolves "other.xxx". So im not sure (understand) what you problem is

> CONFIG_HOSTNAME=3cx.eu

that is a domain

**Reply by joerg-wille** [reply]
I have a different use case, but changing the hostname seems to be what I would like to do. I have connected 2 Fritz!boxes via VPN and I would like to have separate search domains to solve DNS issues. So I want one the boxes to use the "fritz1.box" instead "fritz.box". Other than that I actually do not want to change anything at all. I could not find a way to change the domain on the regular Weg-IF, nor was I able to modify it using Fritz!Box Tools. I guess it must be possible using Freetz. What is the easiest way to this? Could not find any patch or similar. Thanks!

**Comment by fraternl** [comment]
It is not my goal to rename the domain.
My one and only goal is to make a situation where the new prerequisites of 3CX are followed.
I actually only need one single host override.

If I choose the domain frater.3cx.eu at the 3CX-company **they** will make sure it resolves to the WAN-ip of the Fritzbox' WAN-connection (through DDNS).
When I resolve the domain frater.3cx.eu it **should** however resolve to the 3CX-server on premises (_for example 192.168.188.10_).
This is a new guideline and in future versions the provisioning of telephony devices using an IP will be scratched.

The only way I found to make the Fritzbox' DNS server to do this is by changing the CONFIG_HOSTNAME to 3cx.eu
If I rename the hostname of the local 3CX-server to "**frater**" it will resolve  frater.3cx.eu to 192.168.188.10 inside the LAN and to an Internet IP outside the LAN.

That would be all fine, if it would still be possible to resolve other valid 3cx.eu addresses.
It does NOT, so it isn't really a solution to my problem.
Unless I don't care about that collateral damage.

A host override is a normal option of **dnsmasq** and I used it before on other systems.
I don't know the implications of using **dnsmasq** instead of **multid**.

The Fritzboxes have a lot of benefits, but also some downsides. One of them is DNS and the other is the firewall.
We use pfsense if we need a more mature router.
They are more expensive and need separate devices for modem, wlan, dect, telephony

**Comment by fda77** [comment]
I have no idea how to do this with only AVM functions - except to maintain on every(!) device a hosts file with a special entry for your 3cx-hostname. But to prevent this a dns server was developed :)

**Comment by fraternl** [comment]
...except to maintain on every(!) device a hosts file with a special entry for your 3cx-hostname. But to prevent this a dns server was developed

Even that would not be possible.
These devices are VoIP telephony devices.
Like you said...  DNS was developed for this.

**Comment by fraternl** [comment]
I merely hoped someone knew of some way.
I'm still considering to do it whenever that time comes including the collateral damage.

Somehow the domain 3cx.nl went obsolete several years ago.
Only a few of those domains are still in service because they applied before that. I could do it for the 3cx-servers using 3cx.nl (not 3cx.eu)

**Comment by fraternl** [comment]
What's the consensus regarding dnsmasq?
I know the package itself is outstanding, but how does it fit in?

**Comment by fda77** [comment]
If you dont have a reason by yourself to use it, you dont need it :)

**Comment by fraternl** [comment]
I don't understand...

The reason would be the hostname override.
But only needed for a small percentage of the boxes.

I don't think I will go that route. I will probably select another router for these setups.
